tests/gameboard.test.js

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-plusplus */
+/* eslint-disable no-undef */
+import Gameboard from '../src/modules/data/gameboard';
+
+test('returns false if board position is invalid', () => {
+  const playerBoard = Gameboard();
+  playerBoard.placeShip(playerBoard.ships[0], 0, playerBoard.data, 'down');
+  expect(playerBoard.data[0]).not.toBe(null);
+});
```

Comment:
**[ REQUIRED ]:** We will have to change this test. But worry not I will try to break apart the observations one by one to make it as clear as possible.
- First let check on what you were trying to check: it should return false if board position is invalid, you provide an instance of your gameboard, you pass the placeship method and then you write your assertions checkig nor a not null. When in your test you are explicitly saying that it should return false if position is null. This should be a test for your validatePlacemnet() which i think is not tested. See  that validates placement is whats taking care of determining if certain positios is valid or not. You should test that. With this logic, you could test if the Placement took place for example.